File: Java/src/main/java/org/codecop/lcdnumbers/DigitFactory.java
package org.codecop.lcdnumbers;

import java.util.Objects;

/**
 * Creates (LCD) digits from patterns.
 */
public class DigitFactory {

    private final Patterns patterns;

    public DigitFactory(Patterns patterns) {
        Objects.requireNonNull(patterns);

        this.patterns = patterns;
    }

    public Digit create(int digit) {
        return new Digit(digit, patterns);
    }
}

File: Csharp-MsDependencyInjection/LcdNumbers/DigitFactory.cs
namespace Org.Codecop.Lcdnumbers
{
    /// <summary>Creates (LCD) digits from patterns.</summary>
    public class DigitFactory : IDigitFactory
    {
        private readonly IPatterns patterns;

        public DigitFactory(IPatterns patterns)
        {
          this.patterns = patterns;
        }

        public virtual Digit Create(int digit)
        {
            return new Digit(digit, patterns);
        }
    }
}
